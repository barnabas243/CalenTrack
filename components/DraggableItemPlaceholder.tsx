// Import necessary components from React Native
import {View, StyleSheet} from 'react-native';
import React from 'react';
import {useTheme} from 'react-native-paper';

const DraggableItemPlaceholder = () => {
  const {colors} = useTheme();

  const colorsStyle = {
    backgroundColor: colors.surfaceVariant,
    borderColor: colors.primary, // Border color
    shadowColor: colors.shadow,
    wOffset: {width: 0, height: 2}, // Shadow offset
    shadowOpacity: 0.3, // Shadow opacity
    shadowRadius: 4, // Shadow radius
  };

  return <View testID="draggable-item-placeholder" style={[styles.container, colorsStyle]} />;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginVertical: 8, // Added vertical margin for better spacing
    padding: 16, // Padding to make it look like a card
    borderRadius: 8, // Rounded corners
    borderWidth: 1, // Border width
    elevation: 2, // Elevation for Android shadow
  },
});

export default DraggableItemPlaceholder;
