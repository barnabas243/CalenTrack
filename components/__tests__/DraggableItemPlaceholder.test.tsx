import React from 'react';
import {render} from '@testing-library/react-native';
import {ThemeProvider, DefaultTheme} from 'react-native-paper';
import DraggableItemPlaceholder from '../DraggableItemPlaceholder'; // Adjust the import path as needed
import {StyleSheet} from 'react-native';

describe('DraggableItemPlaceholder Component', () => {
  const mockTheme = {
    ...DefaultTheme,
    colors: {
      ...DefaultTheme.colors,
      surfaceVariant: '#f0f0f0',
      primary: '#6200ee',
      shadow: '#000000',
    },
  };

  const wrapper = ({children}: any) => <ThemeProvider theme={mockTheme}>{children}</ThemeProvider>;

  it('renders correctly', () => {
    const {toJSON} = render(<DraggableItemPlaceholder />, {wrapper});
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies correct styles based on theme', () => {
    const {getByTestId} = render(<DraggableItemPlaceholder />, {wrapper});

    const view = getByTestId('draggable-item-placeholder');

    const style = StyleSheet.flatten(view.props.style);

    // Check background color
    expect(style.backgroundColor).toBe(mockTheme.colors.surfaceVariant);
    // Check border color
    expect(style.borderColor).toBe(mockTheme.colors.primary);
    // Check shadow color
    expect(style.shadowColor).toBe(mockTheme.colors.shadow);
    // Check other styles like shadowOpacity, shadowRadius, etc.
    expect(style.shadowOpacity).toBe(0.3);
    expect(style.shadowRadius).toBe(4);
  });
});
