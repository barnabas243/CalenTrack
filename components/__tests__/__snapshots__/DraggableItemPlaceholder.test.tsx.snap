// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableItemPlaceholder Component renders correctly 1`] = `
<View
  style={
    [
      {
        "borderRadius": 8,
        "borderWidth": 1,
        "elevation": 2,
        "flex": 1,
        "marginVertical": 8,
        "padding": 16,
      },
      {
        "backgroundColor": "#f0f0f0",
        "borderColor": "#6200ee",
        "shadowColor": "#000000",
        "shadowOpacity": 0.3,
        "shadowRadius": 4,
        "wOffset": {
          "height": 2,
          "width": 0,
        },
      },
    ]
  }
  testID="draggable-item-placeholder"
/>
`;
